web/modules/custom/hoeringsportal_citizen_proposal/src/Plugin/Block/CitizenProposalSupportCounter.php

```diff
@@ -56,16 +59,15 @@ public function getCacheMaxAge() {
    */
   public function build() {
     $node = $this->routeMatch->getParameter('node');
-
-    $supportCount = $this->connection->select('hoeringsportal_citizen_proposal_support')
-      ->condition('node_id', $node->nid->value ?? 0)
-      ->countQuery()
-      ->execute()
-      ->fetchField();
+    $supportCount = $this->helper->getProposalSupportCount($node->nid->value ?? 0);
+    $supportPercentage = $supportCount ? $this->helper->calculateSupportPercentage((int) $supportCount) . '%' : '0%';
```

Comment:
See comments on the same code in the `Helper` class.

Reply:
Fixed